src/main/java/tech/ypsilon/bbbot/database/codecs/VerificationCodec.java

```diff
@@ -0,0 +1,82 @@
+package tech.ypsilon.bbbot.database.codecs;
+
+import com.mongodb.client.MongoCollection;
+import org.bson.BsonReader;
+import org.bson.BsonWriter;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.types.ObjectId;
+import tech.ypsilon.bbbot.database.MongoController;
+import tech.ypsilon.bbbot.database.structs.VerificationDocument;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.Random;
+
+/**
+ * @author Christian Schliz
+ * @version 1.0
+ */
+public class VerificationCodec implements Codec<VerificationDocument> {
+
+    private static final String MONGO_COLLECTION = "Verification";
+    private static final int VERIFICATION_CODE_LENGTH = 3;
+
+    private static final String FIELD_ID = "_id";
+    private static final String FIELD_USER_ID = "userId";
+    private static final String FIELD_STUDENT_CODE = "studentCode";
+    private static final String FIELD_VERIFICATION_CODE = "verificationCode";
+    private static final String FIELD_VERIFIED = "verified";
+    private static final String FIELD_EMAIL_LAST_SENT = "emailLastSent";
+
+    private static String generateVerificationCode() {
+        int number = new Random().nextInt((int) Math.pow(10, VERIFICATION_CODE_LENGTH));
+        StringBuilder builder = new StringBuilder(number);
+        while (builder.length() < VERIFICATION_CODE_LENGTH)
+            builder.insert(0, "0");
+        return builder.toString();
+    }
+
+    public static VerificationDocument insert(Long userId, String studentCode) {
+        VerificationDocument verificationCodec = new VerificationDocument(new ObjectId(), userId, studentCode,
+                generateVerificationCode(), false, Date.from(Instant.now()));
+
+        getCollection().insertOne(verificationCodec);
```

Comment:
Maybe you also want to delete all old occurences of the VerificationDocument to make sure there is allways only one with the same id/userid in the database, in case something weird happens.

Reply:
done: d63611d